src/webview/memory-webview-view.tsx

```diff
@@ -89,6 +93,15 @@ class App extends React.Component<{}, MemoryAppState> {
         messenger.sendNotification(readyType, HOST_EXTENSION, undefined);
     }
 
+    public componentDidUpdate(_: {}, prevState: MemoryAppState): void {
+        const addressPaddingNeedsUpdate =
+            (this.state.addressPadding === 'Min' && this.state.memory !== prevState.memory)
+            || this.state.addressPadding !== prevState.addressPadding;
+        if (addressPaddingNeedsUpdate) {
+            this.setState({ effectiveAddressLength: this.getEffectiveAddressLength(this.state.memory), });
+        }
```

Comment:
Probably React optimizes away an update for us if we set a state value to the same as its current value, but we could be conservative here and compute the effective address length and only set state if the computed value differs from the current value.

Reply:
Good idea! As far as I know React avoids triggering a re-rendering if the state is unchanged. But it needs to do a shallow comparison of the entire state, which is a more expensive than just testing here whether the single `number` is different. I've added a guard.